docs/source/installation.rst

```diff
@@ -211,3 +211,15 @@ These dependencies are managed, installed and referenced by the library using `P
 No additional dependencies should be needed after running the pip install.
 
 .. _Poetry: https://python-poetry.org/docs/
+
+Version compatibility
+=====================
+
+The Mesh Python SDK will perform the version compatibility check when connecting to the Mesh server.
+It will ask the server for its version number and will validate it according to the rules
+found in the `Versions <https://volue-public.github.io/energy-mesh-python/versions.html>` section.
+
+The Mesh server will also perform the version compatibility check based on the version
+sent by the Mesh Python SDK. For the version metadata to be correctly populated,
+the `volue.mesh` package should be installed using the
+`recommended procedure <https://volue-public.github.io/energy-mesh-python/installation.html#setup-for-users-recommended>`.
```

Comment:
Maybe it is worth to mention how is it sent (via gRPC metadata) and what will happen if you run the lib from source?